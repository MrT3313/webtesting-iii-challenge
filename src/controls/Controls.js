import React from 'react';

const Controls = props => {
  const { locked, closed, toggleLocked, toggleClosed } = props;

  return (
    <div className="controls panel">
      <button
        data-testid = 'tgl_lockGate'
        disabled={!closed} 
        onClick={toggleLocked} 
        className="toggle-btn"
      >
        {locked ? 'Unlock Gate' : 'Lock Gate'}
      </button>
      <button
        data-testid = 'tgl_openGate'
        disabled={locked} 
        onClick={toggleClosed} 
        className="toggle-btn"
      >
        {closed ? 'Open Gate' : 'Close Gate'}
      </button>
    </div>
  );
};

export default Controls;
