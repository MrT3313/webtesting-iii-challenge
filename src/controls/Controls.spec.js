// Test away!
import React from 'react'
import renderer from "react-test-renderer";
import "jest-dom/extend-expect";

import { render, fireEvent, getByTestId } from "@testing-library/react";

import Controls from "./Controls.js";
import Dashboard from "../dashboard/Dashboard";

// import "@testing-library/react/cleanup-after-each";


describe("<Controls />", () => {
    
    it("should match snapshot", () => {
        const tree = renderer.create(<Controls />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('testing controls', () => {
        const { getByTestId } = render(<Dashboard />)
        const toggleLock = getByTestId('tgl_lockGate')
        const toggleOpen = getByTestId('tgl_openGate')

        // Unlocked & Open
        expect(toggleLock).toHaveTextContent(/lock gate/i)
        expect(toggleLock).toBeDisabled()
        expect(toggleOpen).toHaveTextContent(/close gate/i)
        expect (toggleOpen).not.toBeDisabled()

        // Unlocked & Closed 
        fireEvent.click(toggleOpen)
        expect(toggleOpen).toHaveTextContent(/open gate/i)
        expect(toggleOpen).not.toBeDisabled()
        expect(toggleLock).toHaveTextContent(/lock gate/i)
        expect(toggleLock).not.toBeDisabled()

        // Locked & Closed
        fireEvent.click(toggleLock)
        expect(toggleOpen).toHaveTextContent(/open gate/i)
        expect(toggleOpen).toBeDisabled()
        expect(toggleLock).toHaveTextContent(/unlock gate/i)
        expect(toggleLock).not.toBeDisabled()
    })

    it('Mock Functions', () => {
        const tgl_Lock_Mock  = jest.fn()
        const tgl_Open_Mock = jest.fn()

        tgl_Lock_Mock()
        expect(tgl_Lock_Mock).toHaveBeenCalled()
        
        tgl_Lock_Mock('TACO')
        expect(tgl_Lock_Mock).toHaveBeenCalledWith('TACO')

        tgl_Open_Mock()
        expect(tgl_Open_Mock).toHaveBeenCalled()
        
        tgl_Open_Mock('MANY TACOS')
        expect(tgl_Open_Mock).toHaveBeenCalledWith('MANY TACOS')
    })
})