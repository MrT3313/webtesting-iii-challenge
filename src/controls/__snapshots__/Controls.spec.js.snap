// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> should match snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="tgl_lockGate"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="tgl_openGate"
  >
    Close Gate
  </button>
</div>
`;
